src/jvm/main/org/jetbrains/kotlinx/lincheck/strategy/managed/ManagedStrategy.kt

```diff
@@ -1702,4 +1839,13 @@ private const val OBSTRUCTION_FREEDOM_LOCK_VIOLATION_MESSAGE =
     "The algorithm should be non-blocking, but a lock is detected"
 
 private const val OBSTRUCTION_FREEDOM_WAIT_VIOLATION_MESSAGE =
-    "The algorithm should be non-blocking, but a wait call is detected"
\ No newline at end of file
+    "The algorithm should be non-blocking, but a wait call is detected"
+
+/**
+ * With idea plugin enabled, we should not use default Lincheck timeout
+ * as debugging may take more time than default timeout.
+ */
+private const val DEBUGGER_TIMEOUT = 1000L * 60 * 60 * 24 * 365
```

Comment:
Shouldn't it be just "INFINITE_TIMEOUT" here? The reason you need this is to not to fail by timeout when the execution is paused in the debugger.